internal/sql/pg/system.go

```diff
@@ -624,26 +559,17 @@ func columnInfo(ctx context.Context, conn *pgx.Conn, schema, tbl string) ([]ColI
 	return colInfo, nil
 }
 
-// ColumnInfoer is satisfied by a type that is capable of describing tables.
-// This permits testing higher level packages with a stub DB.
-type ColumnInfoer interface {
-	ColumnInfo(ctx context.Context, schema, tbl string) ([]ColInfo, error)
-}
-
 // ColumnInfo attempts to describe the columns of a table in a specified
-// PostgreSQL schema.
+// PostgreSQL schema. The results are **as reported by information_schema.column**.
 //
 // If the provided sql.Executor is also a ColumnInfoer, its ColumnInfo method
 // will be used. This is primarily for testing with a mocked DB transaction.
 // Otherwise, the Executor must be one of the transaction types created by this
 // package, which provide access to the underlying DB connection.
 func ColumnInfo(ctx context.Context, tx sql.Executor, schema, tbl string) ([]ColInfo, error) {
```

Comment:
Exported functions don't take any `ColInfo`, only return for general information purposes.  Docs for `ColInfo` elaborate.  This `ColumnInfo` won't be need needed outside of the `pg` packages, but I find it generally useful, so didn't go so far as to unexport it or `ColInfo`, but it wouldn't derail any of the work in engine or elsewhere.